### PR TITLE
Button to login and backend architecture to logout

### PR DESCRIPTION
**Pull request by saprap1**
## Description
Allows CampusMap users to login with RPI CAS authentication using a login button on the front page. Users will be directed to the official CAS login page and then redirected back to CampusMap once a successful authentication is complete. Users can check their authentication status under the Profile tab. Logout capability is also part of this branch, but it requires going to the specific logout endpoint.

Fixes #(issue name)

## Solution
Created a Express session to keep track of user login information. Utilized the cas-authentication module for Node to communicate with RPI's CAS server. Set up a controller and route in the backend to handle login/logout/authenticate requests. The front end consists of adding a button to the header that communicates with the backend routes.

## Known Issues
None

## Testing Procedures
Try to login using the button and your CAS credentials. Check your login username is stored by clicking Profile. Then go to the api/cas/logout endpoint to see that you have been successfully logged out. Return to the CampusMap homepage and click on profile to double check that your session ended and you are no longer authenticated.

## Checklist for author
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if any)
- [x] My changes generate no new warnings

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -29,13 +29,18 @@
...
     "express": "^4.17.1",
+    "express-cas-authentication": "^0.1.0",
+    "express-session": "^1.17.1",
     "mongodb": "^3.2.3",
     "mongoose": "^5.8.1",
     "morgan": "^1.10.0",
-    "typescript": "^3.8.3"
+    "typescript": "^3.8.3",
+    "passport": "^0.4.1",
+    "passport-local": "^1.0.0"
```
Comment (gaskij) [inline api/package.json:43]: We can probably yeet these, right? Do we use it for anything?

File: api/src/routes/cas.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+} from '../controllers/cas';
+
+/** Instantiate CASAuthentication */
+const cas = new CASAuthentication({
+  /* eslint-disable-next-line @typescript-eslint/camelcase */
+  cas_url: 'https://cas-auth.rpi.edu/cas',
+  /* eslint-disable-next-line @typescript-eslint/camelcase */
+  service_url: process.env.NODE_ENV === 'production' ? 'https://rpicampusmap.herokuapp.com' : 'http://localhost:3000',
+  /* eslint-disable-next-line @typescript-eslint/camelcase */
+  session_name: 'casUser',
```
Comment (gaskij) [inline api/src/routes/cas.ts:21]: You can do `/* eslint-disable @typescript-eslint/camelcase */` one line 16 and then `/* eslint-enable @typescript-eslint/camelcase */` after the `session_name` property to make this cleaner

**Review by gaskij — CHANGES_REQUESTED** [review]

File: api/package.json (hunk 1 above)
Comment (saprap1) [reply api/package.json:43]: indeed that was going to be another attempt

**Review by stadtc — APPROVED** [review]

**Review by gaskij — APPROVED** [review]
lgtm

**Pull request merged**